platform/jobs/pipelines/edxPlatformPythonPipelinePr.groovy

```diff
@@ -0,0 +1,63 @@
+package platform
+
+import org.yaml.snakeyaml.Yaml
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_TEAM_SECURITY
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+
+/* stdout logger */
+Map config = [:]
+Binding bindings = getBinding()
+config.putAll(bindings.getVariables())
+PrintStream out = config['out']
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map ghprbMap = [:]
+try {
+    out.println('Parsing secret YAML file')
+    String ghprbConfigContents = new File("${GHPRB_SECRET}").text
+    Yaml yaml = new Yaml()
+    ghprbMap = yaml.load(ghprbConfigContents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+// This is the job DSL responsible for creating the main pipeline job.
+pipelineJob('edx-platform-python-pipeline-pr') {
+
+    definition {
+
+        triggers {
+            pullRequest {
+                admins(ghprbMap['admin'])
+                useGitHubHooks()
+                triggerPhrase(/.*jenkins\W+run\W+pipeline\W+python.*/)
+                onlyTriggerPhrase(true)
+                userWhitelist(ghprbMap['userWhiteList'])
+                orgWhitelist(ghprbMap['orgWhiteList'])
+                extensions {
+                    commitStatus {
+                        context('jenkins/python-pipeline')
+                    }
+                }
+            }
+        }
+
+        cpsScm {
+            scm {
+                git {
+                    remote {
+                        credentials('jenkins-worker')
+                        github('edx/edx-platform', 'ssh', 'github.com')
+                        refspec('+refs/heads/*:refs/remotes/origin/* +refs/pull/*:refs/remotes/origin/pr/*')
+                        branch('\${sha1}')
+                    }
+                }
+            }
+            scriptPath('Jenkinsfile')
```

Comment:
Are we ultimately just going to have a single Jenkinsfile containing the pipeline(s) for all test types and quality checks, or are we going to need multiple Jenkinsfiles?  If the latter, we may want to group them in a directory with appropriate names.

Reply:
I'm not really sure yet.

As we do more discovery in areas such as the external workspace plugin... we may find ways to cut down CPU time by avoiding redundant steps (like a git checkout). To me it could make sense to have one file if there's numerous shared steps like that.. but if not, splitting them out could make it easier to follow.

We can always change how this is organized as we add more complexity too. What do you think?

Reply:
Sure, we can keep it this way for now to match the usual convention and then change it later if appropriate.